### PR TITLE
Added a new "adjust_total_staking_currency" extrinsic

### PR DESCRIPTION
**Pull request by syan095**
Added unit test and updated the benchmark

### CODE REVIEW

File: modules/homa-lite/src/lib.rs [hunk 1]
```diff
@@ -224,6 +232,29 @@ pub mod module {
...
 
+		/// Adjusts the total_staking_currency by the given difference.
+		/// Requires `T::GovernanceOrigin`
+		///
+		/// Parameters:
+		/// - `adjustment`: The difference in amount the total_staking_currency should be adjusted
+		///   by.
+		#[pallet::weight(< T as Config >::WeightInfo::adjust_total_staking_currency())]
+		#[transactional]
+		pub fn adjust_total_staking_currency(origin: OriginFor<T>, staking_total: BalanceAdjustment) -> DispatchResult {
```
Comment (xlc) [inline modules/homa-lite/src/lib.rs:243]: we already have `Amount` type, which is signed version of `Balance`

https://github.com/AcalaNetwork/Acala/blob/1915058c55bf76661cbd6af9ac367abc0099dccf/modules/currencies/src/lib.rs#L174

File: modules/homa-lite/src/lib.rs [hunk 2]
```diff
@@ -54,7 +51,8 @@ pub mod module {
 		type WeightInfo: WeightInfo;
 
 		/// Multi-currency support for asset management
-		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>;
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>
```
[the remark this answers is not in the document]
Comment (xlc) [reply modules/homa-lite/src/lib.rs:57]: not resolved?

[the remark this answers is not in the document]
Comment (syan095) [reply modules/homa-lite/src/lib.rs]: Sorry my bad - I misread the comment. THis should be fixed now. I needed  "use" both MultiCurrency and MultiCurrencyExtended.

**Review by xlc — DISMISSED** [review]

File: modules/homa-lite/src/lib.rs [hunk 3]
```diff
@@ -93,6 +99,8 @@ pub mod module {
 		MintAmountBelowMinimumThreshold,
 		/// The amount of Staking currency used has exceeded the cap allowed.
 		ExceededStakingCurrencyMintCap,
+		/// The adjustment to total_staking_currency causes arithmatic error
+		InvalidAdjustmentToTotalStakingCurrency,
```
Comment (shaunxw) [inline modules/homa-lite/src/lib.rs:103]: Use `ArithmecError`

**Pull request merged**